Disable service account key deletion

b01901143 opened the pull request:
Temporarily disable deletion of old service account keys.
Preparation for deploying to prow for testing.

b01901143 committed:
Disable service account key deletion

Added flag to enable/disable svc-key deletion.
Added flag to switch between one-shot and continuous mode for
secret-rotator.
Temporarily default to disabling deletion of old service account keys.
Preparation for deploying to prow for testing.

spiffxp commented on secret-rotator/svckey/svckey.go:
```diff
@@ -92,10 +92,11 @@ func (p *Provisioner) Deactivate(labels map[string]string, version string) error
-	if err != nil {
-		return fmt.Errorf("Projects.ServiceAccounts.Keys.Delete: %v", err)
-	}
+	// temporarily disabling service account key deletion, for safety reasons.
+	// _, err := p.Service.Projects.ServiceAccounts.Keys.Delete(name).Do()
+	// if err != nil {
+	// 	return fmt.Errorf("Projects.ServiceAccounts.Keys.Delete: %v", err)
+	// }
```
Can you gate this behind a flag? e.g. plumb through NewProvisioner?

b01901143 replied:
I added a flag `--enable-deletion`  to controller this, which is default to false for now.

BenTheElder commented on secret-rotator/svckey/svckey.go:
```diff
@@ -47,18 +47,24 @@ func (svc ServiceAccountKeySpec) Labels() map[string]string {
 }
 
-func NewProvisioner() (*Provisioner, error) {
+func NewProvisioner(flag bool) (*Provisioner, error) {
```
this variable name should be more descriptive and there should be a godoc comment on this exported method

b01901143 replied:
renamed to `enableDeletion`, and added godoc comments.

BenTheElder commented on secret-rotator/svckey/svckey.go:
```diff
@@ -47,18 +47,24 @@ func (svc ServiceAccountKeySpec) Labels() map[string]string {
-	Service *iam.Service
+	// if EnableDeletion is set to true, the provisioner deletes the old svc key of 'version'
+	// when Deactivate('labels', 'version') is called.
+	EnableDeletion bool
```
is there a reason for this to be public yet also a constructor parameter?

b01901143 replied:
I think I'll make it private, since it wouldn't be accessed outside of this struct.

BenTheElder commented on secret-rotator/svckey/svckey.go:
```diff
@@ -92,12 +101,16 @@ func (p *Provisioner) Deactivate(labels map[string]string, version string) error
-	klog.V(2).Infof("Deactivated ver. %s: %s ", version, name)
+		klog.V(2).Infof("Deactivated ver. %s: %s ", version, name)
+	} else {
+		klog.V(2).Infof("Should deactivate ver. %s: %s ", version, name)
```
Maybe roughly "did not deactivate ver. because"? Should is not quite as telling

b01901143 replied:
That's true. Maybe change to `klog.V(2).Infof("Did not deactivate ver. %s: %s, because flag --enable-deletion is not set", version, name)` to point to the `--enable-deletion` flag?

BenTheElder commented:
/lgtm
/hold

b01901143 committed:
Modify deactivation info message

In the case that --enable-deletion is not set, the deactivation info
message should inform the user that the keys were not deleted because of
this flag.

spiffxp reviewed (approved):
/approve
/lgtm
/hold cancel
